evap/evaluation/models.py

```diff
@@ -897,7 +897,13 @@ def can_staff_delete(self):
         states_with_votes = ["in_evaluation", "reviewed", "evaluated", "published"]
         if any(course.state in states_with_votes and not course.is_archived for course in self.courses_participating_in.all()):
             return False
-        return not (self.is_contributor or self.is_grade_publisher or self.is_staff or self.is_superuser)
+        if self.is_contributor or self.is_grade_publisher or self.is_staff or self.is_superuser:
+            return False
+        if any(not user.can_staff_delete() for user in self.represented_users.all()):
+            return False
+        if any(not user.can_staff_delete() for user in self.ccing_users.all()):
+            return False
```

Comment:
this might imply additional queries per user in the user index view. i'll check and update the respective test